### PR TITLE
Enable `println` warning

### PR DESCRIPTION
**Pull request by davidpdrsn**
Turns out there is a clippy lint for usage of `println!`. We should probably enable that.

### CODE REVIEW

File: axum/src/routing/mod.rs [hunk 1]
```diff
@@ -106,6 +106,8 @@ where
     /// Unless you add additional routes this will respond with `404 Not Found` to
     /// all requests.
     pub fn new() -> Self {
+        println!("prints not allowed. Build should fail");
```
Comment (davidpdrsn) [inline axum/src/routing/mod.rs:109]: Gonna verify that this causes CI to fail.

**Comment by davidpdrsn** [comment]
We have that enabled as well but probably could be allowed in tests.

CI didn't fail though so have to tweak something.

**Comment by jplatte** [comment]
Does it fail with `dbg!`? Maybe you "just" have to update your muscle memory to that? 😅

**Comment by davidpdrsn** [comment]
`dbg!` didn't fail CI either because we deny warnings. But I think the setup I have now is pretty decent. `dbg!` and `println!` are just warnings locally, but always allowed in tests. Any outside of tests will make CI fail.

I do like having `dbg!` and `println!` left in tests for easy debugging if a test starts failing.

**Review by jplatte — APPROVED** [review]

**Pull request merged**